### DISCUSSION TITLE
aria-labelledby in select referring to both label, and selected value elements

### DISCUSSION THREAD

**Discussion by alirezamirian** [adobe/react-spectrum#3821] [General]
Not sure if it's an expected behaviour or not, but the select is specified as labelled by both the label element and the value element:
![image](https://user-images.githubusercontent.com/3150694/206186725-2bcb429c-ce3f-4ea4-af68-7fb82ebdda77.png)

![image](https://user-images.githubusercontent.com/3150694/206187143-5db38e1c-0ca9-47f1-9a31-4b99bb9fc5cc.png)

It's causing issues when selecting a react-aria-powered select element with [playwright's role selector](https://playwright.dev/docs/selectors#role-selector) (`page.getByRole('button', { name: 'the label' })`). Maybe it's a bug in playwright's selector, and not sure if multiple aria-labelledby is allowed. But in any case, it seems a little weird to label the select by the selected value. Perhaps it was done so to have the select labeled by the placeholder value, which still can be questionable.
If it's not by design, I can open an issue (and maybe PR) to fix it.

**Comment by LFDanLu** [comment]
The reason we have multiple values in this `aria-labelledby` here is to ensure screen readers announce both the label of the select and the current selected value. Without the id of the span, the title of the select is the only thing that would be announced upon tabbing to the select, and the user would have to open the drop down to focus the selected option to hear it announced.

**Reply by snowystinger** [reply]
You could probably make use of the Playwright regex matching for names. See the example in the link you shared: `role=button[name=/Click( me)?/]`

**Reply by alirezamirian** [reply]
Using regex actually worked. Thanks.
